Merge back to master

ironjan opened the pull request.

Commit:
iss-hipeb-9 #comment It works!
Testing smart commits

Commit:
HIPEB-9 #comment It works!
Testing smart commits, the 2nd

Commit:
HIPEB-11 Testing Jenkins build

Commit:
HIPEB-11 testing jenkins builds

Commit:
HIPEB-11 testing jenkins builds

mwever committed:
Initialized project seed for setting up testing environment.

mwever committed:
reloaded the project from scratch

mwever committed:
removed complete seed from project again to set up a new project seed

mwever committed:
Merge branch 'develop' of git.cs.upb.de:HiP/HiPCMS into develop

mwever committed:
deleted all project files again

mwever committed:
New added project seed, working with scoverage, providing testing and coverage reporting.

mwever committed:
Added license and readme again to project

mwever committed:
added index page design.

mwever committed:
Added angularjs as nodejs module

uxxman committed:
HIPCMS-222 - Removing old files

uxxman committed:
HIPCMS-222 - Adding gitignore file specific to Play Scala project

uxxman committed:
HIPCMS-222 - Adding new Play Scala project

uxxman committed:
HIPCMS-222 - Initializing porject with bower for managing front-end dependencies

uxxman committed:
HIPCMS-222 - Adding asset compilation configurations for '.less' files

uxxman committed:
HIPCMS-222 - Adding angularjs as bower component

uxxman committed:
HIPCMS-222 - Adding bootstrap as bower component

uxxman committed:
HIPCMS-222 - Adding angular-ui-router as bower component

uxxman committed:
HIPCMS-222 - Creating basic structure for angular app and stylesheets

Commit:
Merge branch 'iss-hipcms-222-new-project-template' into 'develop'

Iss hipcms 222 new project template

New Play!Scala template with angular and basic structure included

See merge request !1

Commit:
HiPCMS-66 Added project dependency package.json

Commit:
HiPCMS-66 Installed protractor and updated conf file for it

venkateshpenchala committed:
HiPCMS-66: Adding Karma-Jasmine related dependency and also Scala Tests

venkateshpenchala committed:
HiPCMS-66: Adding Protractor test cases for the basic seed.

venkateshpenchala committed:
HiPCMS-66: Configured Karma for Unit Tests.

venkateshpenchala committed:
HiPCMS-66: Changing Template for testing

venkateshpenchala committed:
HiPCMS-66: Changing Template for testing

venkateshpenchala committed:
HiPCMS-66: Changing Template for testing

venkateshpenchala committed:
HiPCMS-66: Changing Template for testing

venkateshpenchala committed:
HiPCMS-66: Changing Template for testing

venkateshpenchala committed:
HiPCMS-66: Changing Scala Test Version

venkateshpenchala committed:
HiPCMS-66: Adding Scala Tests

venkateshpenchala committed:
HiPCMS-66: Comming to test Jenkins Sonar Integration

venkateshpenchala committed:
HiPCMS-66: Implementation of Scala Tests

venkateshpenchala committed:
HiPCMS-66: Implementation of Unit and Functional Testing using Scala Tests

venkateshpenchala committed:
HiPCMS-66: Adding Dependency for Scoverage in the build

Commit:
Merge branch 'iss-hipcms-66' into 'develop'

Set up test environment

See http://jira-hip.cs.upb.de:8080/browse/HIPCMS-66

See merge request !3

uxxman committed:
HIPCMS-222 - Removing support for less and coffee scritp and adding static files folder structure

uxxman committed:
HIPCMS-222 - Adding all configurations needed to run mysql and slick in the project

uxxman committed:
HIPCMS-222 - Moving Play app to folder server

uxxman committed:
HIPCMS-222 - Removing bower and other files not needed in server

uxxman committed:
HIPCMS-222 - Adding angular app as client using yeoman

uxxman committed:
HIPCMS-222 - Configuring grunt to work with play app server

Commit:
Merge branch 'iss-hipcms-222-new-project-template' into 'develop'

Iss hipcms 222 new project template

http://jira-hip.cs.upb.de:8080/browse/HIPCMS-222
For setting up DB on your local machine, see http://jira-hip.cs.upb.de:8090/pages/viewpage.action?pageId=7081452#SettingUpDevelopmentEnvironment(Update)-SetupDatabase(MySQL)


See merge request !10

uxxman committed:
HIPCMS-222 - Removing node modules and client related code in server

venkateshpenchala committed:
HiPCMS-222 - Adding Karma and Protractor Config for Testing

venkateshpenchala committed:
HiPCMS-222 - Added jshint dependent file back to the project

uxxman committed:
HIPCMS-222 - Adding database connection configs

venkateshpenchala committed:
HipCMS - 222 Added Headless browser config and removed phantom

venkateshpenchala committed:
HiPCMS - 222 Added Automatic updation of selenium jar for protractor

venkateshpenchala committed:
HiPCMS - 222 - Added Headless browser HtmlUnit for Functional Test

Commit:
Modified comment to HtmlUnit

ironjan committed:
Merge branch 'develop' into iss-hipcms-222-Project-Template

Commit:
Merge branch 'iss-hipcms-222-Project-Template' into 'develop'

Iss hipcms 222 project template



See merge request !13

ironjan committed:
Added travis.yml

ironjan committed:
Updated travis.yml

ironjan committed:
Removed FunctionalSpec

ironjan committed:
Trying out multi-project build

ironjan committed:
CDing back to root?

ironjan committed:
Split commands

ironjan committed:
Merge remote-tracking branch 'cs.upb.de/iss-hipcms-222-Testing-Environment' into travis-ci

ironjan committed:
Moved installation to install_dependencies.sh

ironjan committed:
Using $PROJECT_ROOT in script

venkateshpenchala committed:
HiPCMS - 222 Removed PhantomJS Dependencies

ironjan committed:
Using local bower in install script

ironjan committed:
after_success step was executed after every single build result

ironjan committed:
Fixed typo

ironjan committed:
Using npm test instead of grunt test

ironjan committed:
Split scripts into single steps

ironjan committed:
Added alias for local grunt

ironjan committed:
Trying something else

ironjan committed:
Adding nodemodules to cache

ironjan committed:
Merge remote-tracking branch 'cs.upb.de/iss-hipcms-222-Testing-Environment' into travis-ci

ironjan committed:
Disabled protractor

ironjan committed:
Added .travis.yml file

ironjan commented:
This was just a test-PR to create a correct .travis.yml before moving the active repository to github - it will be moved to the main repo.

Commit:
Merge branch 'iss-hipcms-281-add-travis.yml' into 'develop'

Added .travis.yml file



See merge request !14

ironjan commented:
Reopened, so that @venkateshpenchala can take a look Art protractor

venkateshpenchala committed:
travis-ci Added Selenium Update to install dependency to test Protractor

venkateshpenchala committed:
Added Selenium Update to install dependency to test Protractor Trial-2

venkateshpenchala committed:
Added Selenium Update to install dependency to test Protractor Trial-3

venkateshpenchala committed:
Added Selenium Update to install dependency to test Protractor Trial-4

venkateshpenchala committed:
Added Selenium Update to install dependency to test Protractor Trial-5

venkateshpenchala committed:
Added Selenium Update to install dependency to test Protractor Trial-6

venkateshpenchala committed:
Added Functional Tests with oneServerPerSuite method

venkateshpenchala commented:
Updated The following:
- Downgraded Protractor and grunt-protractor-runner because of version incompatibility.
- Since Protractor needs a running application to test, I have changed the baseURL in protractor configuration.
- Due to selenium driver problems with travis, Instead of using Browser based testing, I have updated to OneServerPerSuit which doesn't need selenium dependency.

ironjan committed:
Added some badges

venkateshpenchala committed:
Merge branch 'develop' into travis-ci

ironjan committed:
Merge pull request #4 from HiP-App/travis-ci

Travis ci

ironjan committed:
Merge branch 'develop' into xx-add-badges

ironjan committed:
Moved test script from .travis.yml to ci-test.sh

HIPCMS-296 #close Moved test script from .travis.yml to ci-test.sh

ironjan committed:
Merge pull request #3 from HiP-App/xx-add-badges

Added some badges

ironjan committed:
Updated Badges.

ironjan committed:
Merge branch 'develop' into iss-hipcms-296-externalize-ci-test-script

ironjan committed:
Added +x flag to ci-test.sh

ironjan committed:
Merge pull request #6 from HiP-App/xx-update-batches

Updated Badges.

ironjan committed:
Merge branch 'develop' into iss-hipcms-296-externalize-ci-test-script

ironjan committed:
Adapted installation script to work outside Travis CI

HIPCMS-298 #time 45m

ironjan committed:
More readable output for ci-test.sh

ironjan committed:
Added .gitignore for project root

ironjan committed:
Merge pull request #5 from HiP-App/iss-hipcms-296-externalize-ci-test-script

Moved test script from .travis.yml to ci-test.sh

ironjan committed:
Merge remote-tracking branch 'origin/develop' into xx-minor-improvements

ironjan committed:
Merge pull request #8 from HiP-App/xx-minor-improvements

Some minor improvements

ironjan committed:
Added Slack notifications

HIPCMS-300 #time 15m
HIPCMS-300 #closes

ironjan committed:
Merge pull request #9 from HiP-App/iss-hipcms-300

Added Slack notifications

nitishspatkar committed:
Merge pull request #7 from HiP-App/iss-hipcms-298

Adapted installation script to work outside Travis CI

ironjan committed:
Moved split project into own folder

ironjan committed:
Added play-silhouette-angular-seed

ironjan committed:
Added Slick and MySQL to template

ironjan committed:
Removed SocialAuth

ironjan committed:
Added basic travis setup

 * setup.sh    – installs all dependencies
 * test.sh     – executes all tests. Currently just returns 0 for success
 * .travis.yml – for TravisCI

ironjan committed:
Deleted split project

ironjan committed:
Moved play-silhouette-angular-seed to project root

ironjan committed:
Fixed Travis CI scripts

ironjan committed:
Added bower-modules to .gitignore

ironjan committed:
Replaced main.scss by its compilation result main.css

We decided to use Plain-CSS. The css-file added here was the one delivered by the application.

venkateshpenchala committed:
Merge pull request #10 from HiP-App/iss-hipcms-305

Project based on play-silhouette-angular-seed

ironjan committed:
Replaced the forgotten DAOs

 * Login Credentials were saved in memory

ironjan committed:
Added some error logging

ironjan committed:
Removed remaining references to SocialAuth

venkateshpenchala committed:
HiPCMS 306 Adding Scala tests

venkateshpenchala committed:
HiPCMS 306 Deleting existing test configurations

venkateshpenchala committed:
HiPCMS 306 Adding Testing related node dependencies

venkateshpenchala committed:
HiPCMS 306 Added Auto selenium update for e2e tests

venkateshpenchala committed:
HiPCMS 306 Commenting out Ruby and SaSS related settings

venkateshpenchala committed:
HiPCMS 306 Configured Testing related settings

venkateshpenchala committed:
HiPCMS 306 Adding Unit and E2E tests and their resp. config

uxxman committed:
Merge pull request #12 from HiP-App/iss-hipcms-306

Testing Environment for Play Silhouette Angular Seed Project

uxxman committed:
HIPCMS-311 Removed satellizer.js from codebase and added as an external dependency

ironjan committed:
Merge pull request #13 from HiP-App/iss-hipcms-311

HIPCMS-311 Removed satellizer.js from codebase and added as an extern…

ironjan committed:
Revert "HIPCMS-311 Removed satellizer.js from codebase and added as an extern…"

ironjan committed:
Merge pull request #14 from HiP-App/revert-13-iss-hipcms-311

Revert "HIPCMS-311 Removed satellizer.js from codebase and added as an extern…"

ironjan committed:
Merge remote-tracking branch 'origin/develop' into iss-hipcms-310

ironjan committed:
First draft of an updated README.md

ironjan committed:
Explained, why OAuth-classes where kept

ironjan committed:
Added some more explanations and fixed a typo

venkateshpenchala committed:
Merge pull request #16 from HiP-App/iss-hipcms-297

Update README.md

ironjan committed:
Merge remote-tracking branch 'origin/develop' into iss-hipcms-310

ironjan committed:
Removed Gemfiles

ironjan committed:
Removed sass and compass

venkateshpenchala committed:
HiPCMS 308 Added Coday Coverage Configurations

ironjan committed:
Merge pull request #11 from HiP-App/iss-hipcms-310

Forgotten DAOs and some Logging

ironjan committed:
Added UsersController and the corresponding routes

ironjan committed:
Added play-swagger for swagger.json generation

ironjan committed:
Added a Yeoman-Delegator so that swagger.json does not include yeoman.Routes

ironjan committed:
Deleted duplicate Yeoman route

ironjan committed:
Reordered routes and removed unneccessary comments

ironjan committed:
Hiding undocumented routes for now

ironjan committed:
Changed id parameter for updating a single user to UUID as in model

ironjan committed:
Added API documentation for user routes

ironjan commented:
Tried to trigger codacy again.

ironjan committed:
Removed template code from test.sh

ironjan committed:
Added codacy badge

ironjan committed:
Added Travis badge

ironjan committed:
Added function exit_on_error

ironjan committed:
Added database to .travis.yml for FunctionalSpec

ironjan committed:
Added another environment variable to get DATABASE_USER

 * Travis creates a database that is accessible for user "root" or "travis"
 * The current configuration expects a database with user hipcms

ironjan committed:
Removed unnecessary and broken after_script

ironjan committed:
Removed unallowed method error.

 * Documentation is only for GET
 * If you GET that resource, the "Unallowed Method" error will not be thrown

ironjan committed:
Fixed user.routes – routes were generated as, e.g., /api/users/users/

ironjan committed:
Added package for API model definitions

ironjan committed:
Disabled lint warning warn-missing-interpolator

It caused the build to fail because of a false positive in the play-swagger definition:
#        $ref: '#/definitions/controllers.api.protocol.ListUserModel'
was reported as missing, but is used by play-swagger

ironjan committed:
Added model definition for the user admin API

ironjan committed:
Improved naming, added documentation, changed type of user-list parameters

The types of the search and role parameter are now Option[String] instead of
Strings with default values.

venkateshpenchala committed:
Merge pull request #20 from HiP-App/iss-hipcms-308

Codacy Code Coverage, Badges and fixing test.sh

ironjan committed:
Added explicit return types for public methods

ironjan committed:
Applied "Prohibit null literal" and added method documentation

 * https://www.codacy.com/app/HiP-App/HiPCMS/patterns/list#pattern-226

ironjan committed:
Changed return type to Action[JsValue] where applicable

ironjan committed:
Added some documentation

ironjan committed:
Reformatted code with IntelliJ defaults

ironjan committed:
Added explicit type to public method

ironjan committed:
Merge pull request #17 from HiP-App/iss-hipcms-314

API documentation

ironjan committed:
Removed 404 result from /api/users and added 404 results to /api/users/:id

ironjan committed:
Extended UserDAO by findAll and findByEmailOrName

ironjan committed:
Extended UserServiceImpl by retrieveAll and retrieveBySearch

ironjan committed:
Implemented UsersController#userList partially: support search parameter

ironjan committed:
Added retrieve by userID

ironjan committed:
Implemented UsersController#updateUser and renamed: UserResponseModel -> UserProtocolModel

ironjan committed:
Removed empty debug response

ironjan committed:
Extracted function "mapUserToUserProtocolModel"

ironjan committed:
Compacted code

ironjan committed:
Added NotImplementedYet response for role-filter

ironjan committed:
Changed None case to default case

 * If we found a user, everything is fine
 * In all other cases, no user is found

ironjan committed:
Added recovery for IdentityNotFoundException and default cases

ironjan committed:
Merge pull request #22 from HiP-App/iss-hipcms-319-server-side-implementation-of-manage-users-api

WIP: Server Side Implementation of Manage Users API

ironjan committed:
Added deprecation note.

ironjan committed:
Merge pull request #24 from HiP-App/deprecation-note

Added deprecation note.

ironjan commented on app/controllers/CredentialsAuthController.scala:
```diff
@@ -0,0 +1,97 @@
+          Unauthorized(Json.obj("message" -> Messages("invalid.credentials")))
+        }
+      }
+    }.recoverTotal {
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382300/issues/source?bid=3144114&fileBranchId=3144125#l91)

ironjan commented on app/controllers/CredentialsAuthController.scala:
```diff
@@ -0,0 +1,97 @@
+                expirationDateTime = clock.now + c.as[FiniteDuration]("silhouette.authenticator.rememberMe.authenticatorExpiry"),
+                idleTimeout = c.getAs[FiniteDuration]("silhouette.authenticator.rememberMe.authenticatorIdleTimeout")
+              )
+            case authenticator => authenticator
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Lower case pattern matching.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382300/issues/source?bid=3144114&fileBranchId=3144125#l76)

ironjan commented on app/controllers/CredentialsAuthController.scala:
```diff
@@ -0,0 +1,97 @@
+          }
+          case None => Future.failed(new IdentityNotFoundException("Couldn't find user"))
+        }
+      }.recover {
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382300/issues/source?bid=3144114&fileBranchId=3144125#l85)

ironjan commented on app/models/daos/PasswordInfoDAO.scala:
```diff
@@ -0,0 +1,101 @@
+   */
+  def save(loginInfo: LoginInfo, authInfo: PasswordInfo): Future[PasswordInfo] = {
+    val query = loginInfoQuery(loginInfo).joinLeft(slickPasswordInfos).on(_.id === _.loginInfoId)
+    val action = query.result.head.flatMap {
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382315/issues/source?bid=3144114&fileBranchId=3144125#l86)

ironjan commented on app/models/daos/PasswordInfoDAO.scala:
```diff
@@ -0,0 +1,101 @@
+    slickPasswordInfos.filter(_.loginInfoId in loginInfoQuery(loginInfo).map(_.id))
+
+  protected def addAction(loginInfo: LoginInfo, authInfo: PasswordInfo) =
+    loginInfoQuery(loginInfo).result.head.flatMap { dbLoginInfo =>
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382315/issues/source?bid=3144114&fileBranchId=3144125#l31)

ironjan commented on app/models/services/UserServiceImpl.scala:
```diff
@@ -0,0 +1,92 @@
+   * @param user The user to save.
+   * @return The saved user.
+   */
+  def save(user: User) = userDAO.save(user)
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Public method must have explicit type](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382319/issues/source?bid=3144114&fileBranchId=3144125#l60)

ironjan commented on app/modules/SilhouetteModule.scala:
```diff
@@ -0,0 +1,161 @@
+  /**
+   * Configures the module.
+   */
+  def configure() {
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use : Unit = for procedures](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382320/issues/source?bid=3144114&fileBranchId=3144125#l40)

ironjan commented on app/models/daos/DBTableDefinitions.scala:
```diff
@@ -0,0 +1,141 @@
+
+  class UserLoginInfos(tag: Tag) extends Table[DBUserLoginInfo](tag, "userlogininfo") {
+    def userID = column[String]("userID")
+    def loginInfoId = column[Long]("loginInfoId")
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Public method must have explicit type](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382311/issues/source?bid=3144114&fileBranchId=3144125#l51)

ironjan commented on app/models/daos/DBTableDefinitions.scala:
```diff
@@ -0,0 +1,141 @@
+  class Users(tag: Tag) extends Table[DBUser](tag, "user") {
+    def id = column[String]("userID", O.PrimaryKey)
+    def firstName = column[Option[String]]("firstName")
+    def lastName = column[Option[String]]("lastName")
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Public method must have explicit type](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382311/issues/source?bid=3144114&fileBranchId=3144125#l24)

ironjan commented on app/models/daos/DBTableDefinitions.scala:
```diff
@@ -0,0 +1,141 @@
+  val slickOpenIDAttributes = TableQuery[OpenIDAttributes]
+  
+  // queries used in multiple places
+  def loginInfoQuery(loginInfo: LoginInfo) = 
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Public method must have explicit type](https://www.codacy.com/app/HiP-App/HiPCMS/file/2328382311/issues/source?bid=3144114&fileBranchId=3144125#l139)
